ddtrace/tracer/option.go

```diff
@@ -556,3 +586,34 @@ func StackFrames(n, skip uint) FinishOption {
 		cfg.SkipStackFrames = skip
 	}
 }
+
+// parseHTTPCodeRanges parses range pairs and returns a bitset mapping of HTTP status codes.
+func parseHTTPCodeRanges(r string) *bitset.BitSet {
+	re := regexp.MustCompile("\\d{3}(?:-\\d{3})*(?:,\\d{3}(?:-\\d{3})*)*")
+	codes := bitset.New(0)
+	for _, code := range strings.Split(r, ",") {
+		code = strings.TrimSpace(code)
+		if code == "" {
+			continue
+		}
+		if !re.MatchString(code) {
+			log.Warn("Invalid range for %v", code)
+			continue
+		}
+		rg := strings.Split(code, "-")
+		if len(rg) == 1 {
+			val, _ := strconv.Atoi(rg[0])
+			codes.Add(uint(val))
+		} else {
+			if rg[0] > rg[1] {
+				rg[0], rg[1] = rg[1], rg[0]
+			}
+			min, _ := strconv.Atoi(rg[0])
+			max, _ := strconv.Atoi(rg[1])
```

Comment:
My thought was that because `!re.MatchString(code)` validates the input, there wouldn't be a case where `strconv.Atoi()` returns an error. Just added a change where the error would return a log.